src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -3,19 +3,58 @@
     using System;
     using System.Collections.Generic;
     using System.Linq;
+    using System.Net.Http;
     using System.Reflection;
+    using System.Threading.Tasks;
 
     internal static class HeartbeatDefaultPayload
     {
-       public static readonly string[] DefaultFields =
-        {
+        public static readonly string[] DefaultFields =
+         {
             "runtimeFramework",
-            "baseSdkTargetFramework"
+            "baseSdkTargetFramework",
+            "osVersion"
+        };
+
+        public static readonly string[] DefaultOptionalFields =
```

Comment:
We need unittests. Maybe make baseImdsUrl as overridable from UnitTest to mock AzureMetaData, and make sure this class does not crash/handle all responses from the metadata service.
You can use in-process server to mock azure metadata. An example is here:
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/DependencyCollector/NetCore.Tests/DependencyTrackingTelemetryModuleTestNetCore.cs#L190

Reply:
Agreed, I can spend some further time against making some proper unit tests. Thanks for the link to an existing example Cijo!